tests/unit/models/test_model_interfaces.py

```diff
@@ -297,15 +295,17 @@ def __call__(
         (GaussianProcessRegression, _gpr),
         (GaussianProcessRegression, _sgpr),
         (VariationalGaussianProcess, _vgp),
+        (SparseVariational, _svgp),
     ],
 )
 def _gpr_interface_factory(request: Any) -> _ModelFactoryType:
     def model_interface_factory(
         x: TensorType, y: TensorType, optimizer: Optimizer | None = None
-    ) -> GaussianProcessRegression:
+    ) -> tuple[GaussianProcessRegression, Callable]:
         model_interface: type[GaussianProcessRegression] = request.param[0]
         base_model: GaussianProcessRegression = request.param[1](x, y)
-        return model_interface(base_model, optimizer=optimizer)
+        _reference_model: Callable = request.param[1]
```

Comment:
why not initializing with (x,y) immediately here as well?

Reply:
For tests such as `test_gaussian_process_regression_update` we want to compare the interface on updating from data `(x, y)` to `new_data = (x_new, y_new)` with the `update` interface method. To do this it seems to make most sense to initialize the interface with `(x, y)`, then use `model.update(new_data)` and initialize the GPflow reference with `new_data` instead of trying to update the GPflow reference from `(x, y)` to `new_data`. With this setup I need to pass the constructor, instead of an instance. What are your thoughts on this?

Reply:
right, I've noticed that, I think it depends which case you end up using more often - you can still do this even if its an instance, just call the gpr_factory again with new data  - feel free to choose the variant you think is more useful

Reply:
I feel like this way is a bit of a cleaner way of doing it, but if you prefer I can change it to what you suggest, or maybe there's even a way of separating things out so we have a reference model maker that takes the model interface and a dataset and returns the reference model we want?

Reply:
I dont have a strong preference really

Reply:
Ok in that case I'll just leave it